### ISSUE TITLE
Add toggle to move sidebar to left or right side

### ISSUE THREAD

**Issue by e-102** [runelite/runelite#1641]
Within the feature settings you will be given the option to toggle the sidebar to either side of the client or move them (the sidebar options) up into the title bar.

This feature should also allow you to toggle the sidebar options so that it opens in a pop-out window.

**Comment by deathbeam** [comment]
This is just bad idea, its fine as it is now on right side

**Comment by F-Lambda** [comment]
How is this a "bad idea"? Just because something is "fine" doesn't mean options aren't good.

**Comment by Shamus03** [comment]
Here is my argument.  Note the red square - After clicking the wrench to open the sidebar, the wrench MOVES to the right, which requires moving the mouse to a different location to close the bar again.  It would be much more convenient if, through a setting, we could stick the sidebar to the left so the mouse does not have to move:
![sidebar](https://user-images.githubusercontent.com/739815/177674135-8b38a6f6-9a80-49ad-8101-12d3e78d2d51.gif)


It would be much more convenient if the settings button stayed here:
![image](https://user-images.githubusercontent.com/739815/177674752-d0e32d37-a2e2-44ab-80e8-03cb27d85732.png)

**Comment by JamesYZFR1** [comment]
Is there still no intention of adding this feature?

**Comment by Norkotah** [comment]
Why isn't this implemented yet???

**Comment by 0vertime** [comment]
Crazy how all duplicates lead here but the issues been closed for years. Still dropping by once a year to check if its gonna ever happen.

**Comment by DominickCobb-rs** [comment]
Having a standard layout in the client provides an easier option to support the client. Whether this is the intention of the developers, it's a good bit of unnecessary code to implement on a client that will be deprecated in the future, as well as a hassle to help unsavvy users navigate their runelite installation. I'm sure if anyone wanted to contribute this to the client the maintainers would be willing to review it and play ball, but it's not on their horizon and clearly hasn't been for the six years this issue has been open.

**Comment by halfwolf102** [comment]
> This is just bad idea, its fine as it is now on right side

I don't understand how this is a bad idea. Simply saying that comes off as it not having a purpose just because you wouldn't use it yourself. Would it really break things that much? I'm genuinely curious.

**Comment by Robiinn** [comment]
I would love this feature to be implemented for side-by-side clients with an alt. That way the plugin bar of the client on the left would not move over the client on the right side. The current setup would either have an open space between both clients or overlapping plugin bar when extended

**Comment by cdfisher** [comment]
> That way the plugin bar of the client on the left would not move over the client on the right side.

You can use a profile with RuneLite -> Window Settings -> Resize Type set to "Keep window size" to do this already

**Comment by Robiinn** [comment]
> You can use a profile with RuneLite -> Window Settings -> Resize Type set to "Keep window size" to do this already

Not ideal since it makes the game window smaller, but it works for now

**Comment by halfwolf102** [comment]
> I would love this feature to be implemented for side-by-side clients with an alt. That way the plugin bar of the client on the left would not move over the client on the right side. The current setup would either have an open space between both clients or overlapping plugin bar when extended

The game window appears to have a minimum size that it can be. Not sure if it's set by the Runescape client or RuneLite, but either way it's locked at 25% of the screen (50% of the width and 50% of the height). By having the 2 windows next to each other, you're already taking up the entire screen space with just the game on it's own. So the only way to have the RuneLite sidebar visible in addition to this is to have the windows overlap. You can hide the menu with the Plugin Panel Toggle Key (default: CTRL + F11), which should make it to where the windows don't overlap, but I'm guessing you want to keep it up so that probably isn't helpful. As far as I can tell, that seems to be the only option right now. I would suggest that you try to open a ticket for a quality of life / enhancement change though. Assuming it can be done, it's a bit beyond what I'm trying to do, which is just allowing the sidebar to be on the left side of the screen.

**Comment by Robiinn** [comment]
> > I would love this feature to be implemented for side-by-side clients with an alt. That way the plugin bar of the client on the left would not move over the client on the right side. The current setup would either have an open space between both clients or overlapping plugin bar when extended
> 
> The game window appears to have a minimum size that it can be. Not sure if it's set by the Runescape client or RuneLite, but either way it's locked at 25% of the screen (50% of the width and 50% of the height). By having the 2 windows next to each other, you're already taking up the entire screen width with just the game on it's own. So the only way to have the RuneLite sidebar visible in addition to this is to have the windows overlap. You can hide the sidebar with the Plugin Panel Toggle Key (default: CTRL + F11), which should make it to where the windows don't overlap, but I'm guessing you want to keep it up so that probably isn't helpful. As far as I can tell, that seems to be the only option right now. I would suggest that you try to open a ticket for a quality of life / enhancement change though. Assuming it can be done, it's a bit beyond what I'm trying to do, which is just allowing the sidebar to be on the left side of the screen.

I'm playing on a wide screen, so 50% width (and 50% height) minimum is definitely not true for all setups. I also know people who run multiple screens next to each other on a laptop, that I think are smaller than 50%. Having the sidebar on the left for me would be a good solution for 2 clients.

**Comment by halfwolf102** [comment]
> I'm playing on a wide screen, so 50% width (and 50% height) minimum is definitely not true for all setups. I also know people who run multiple screens next to each other on a laptop, that I think are smaller than 50%. Having the sidebar on the left for me would be a good solution for 2 clients.

Ah, okay, that makes sense. In that case, there would probably need to be an interaction added that allows windows to stick together when side by side, similar to how it snaps to the side of the screen. Could maybe even add a toggle for it in the config too (i.e. Multiwindow Snapping or something). Certainly a good idea, not too sure how to implement it though. I'm relatively new to coding java and most of what I've learned has been for this project.

**Comment by kaiba121321-ui** [comment]
Almost 2026 and still waiting on this feature. For me it's a case of the side panel + minimap + inventory, etc. all being on the same side looks really cluttered and I end up with the right half of my screen being entirely UI and the left side being entirely game when the sidebar is open and it's really not a comfortable experience to use honestly. I cannot swap the default OSRS UI layout the opposite direction though... been playing the game since 2004 with inventory, minimap and stuff on the right and chatbox on the left.

**Comment by kaiba121321-ui** [comment]
> This is just bad idea, its fine as it is now on right side

Why on Earth did you ever close this when we're sat here, 7 and a half fucking years later with this STILL being a requested feature. The sheer irony of creating a client to fill out the missing QoL that the base game's client has only to create more QoL issues with the new client is the most absurd thing I've ever seen.

**Comment by cdfisher** [comment]
The "wontfix" label on this suggestion indicates that it's not something we intend to add.

**Comment by kaiba121321-ui** [comment]
> The "wontfix" label on this suggestion indicates that it's not something we intend to add.

Cool,cool, didn't really ask, stated why it's absolutely absurd, hell the amount of thumbs down on the comment before closing it says enough. Tbf I feel like this should be brought to Jagex's attention, like they allow this as an OFFICIAL client for the whole point of it's QoL and you're failing to add basic QoL features. 

(and no, I don't intend to sit here arguing it, I've said what needed to be said, I'm going to get on with my life)

**Comment by OpenCollab-StockholmUni** [comment]
Hi @kaiba121321-ui, we're researchers from Stockholm University studying communication on GitHub.

A [message](https://github.com/runelite/runelite/issues/1641#issuecomment-3447774011) you posted may come across as **rude, disrespectful, or unreasonable** — something that could discourage others from contributing or sharing their ideas:

> why on earth did you ever close this when we're sat here, 7 and a half fucking years later with...

You can opt out of messages like this [here](http://otree-extra-url-d0bb646a8437.herokuapp.com/?hash_id=3b169a73ac5867db6e3edad86ef71ad84bef2137db25adb6aea6841fa4c3f8e5&opt_out=own&source=github).

**If you are the moderator of this repository, you can opt out of messages like this for the entire repo [here](http://otree-extra-url-d0bb646a8437.herokuapp.com/?hash_id=3b169a73ac5867db6e3edad86ef71ad84bef2137db25adb6aea6841fa4c3f8e5&opt_out=all&source=github).**


>[!NOTE]
>Participate in our [four-question survey](http://otree-extra-url-d0bb646a8437.herokuapp.com/?hash_id=3b169a73ac5867db6e3edad86ef71ad84bef2137db25adb6aea6841fa4c3f8e5&opt_out=none&source=github) to provide feedback, and [follow us](https://github.com/huentat/GitHub_Research) for project updates!

---

This message was flagged by an automated system and reviewed using an AI tool to check for tone and clarity concerns.

We’re not here to judge. Our goal is to help make collaboration smoother for everyone.

Messages like this can sometimes cause stress or burnout for others involved in the project.

Here are a few quick tips to make messages clearer and easier to engage with:

- Rephrase strong or absolute language
- Explain your reasoning, or suggest an alternative
- Use tools like AI writing assistants to help reword or check tone

Thanks for contributing to GitHub and helping improve collaboration for everyone!

**Comment by halfwolf102** [comment]
> The "wontfix" label on this suggestion indicates that it's not something we intend to add.

Ah, I thought it was more along the lines of everyone was working on more important aspects such as updated RS content and RL bug fixes, that's unfortunate to hear. Should I close the pull request I made then if that's the case?

**Comment by kaiba121321-ui** [comment]
> > The "wontfix" label on this suggestion indicates that it's not something we intend to add.
> 
> Ah, I thought it was more along the lines of everyone was working on more important aspects such as updated RS content and RL bug fixes, that's unfortunate to hear. Should I close the pull request I made then if that's the case?

That is what I would assume too, but it's starting to seem more like for both Runelite and HDOS that the creators just want to do what they want and do not care for the community that are the entire reason why their client is even a success to begin with.

Hell I wasn't even trying to be an asshole in what I was saying, I was just trying to be straight, to the point and realistic about it... A thread that is closed for 8 years that continuously gets revived and the person who closed it spammed with dislikes... That SCREAMS that the community wants it and we are the reason Runelite even got to the point of being accepted and endorsed by Jagex, without us, the players, Jagex would've banned this client years ago

**Comment by riktenx** [comment]
>  it's starting to seem more like for both Runelite and HDOS that the creators just want to do what they want

Yes, that is how open-source software works. The client was created by Adam. He doesn't get paid to do this, none of the maintainers do. The client and its feature set is quite literally whatever he wants it to be. Your arguments do not matter, the number of people that want this do not matter, whatever you think the logical merit of this feature is or why he's wrong for not wanting to add it, it does not matter. If he does not want to add a feature, it's not happening.

Can someone with the appropriate permissions lock this thread?

**Comment by kaiba121321-ui** [comment]
> > it's starting to seem more like for both Runelite and HDOS that the creators just want to do what they want
> 
> Yes, that is how open-source software works. The client was created by Adam. He doesn't get paid to do this, none of the maintainers do. The client and its feature set is quite literally whatever he wants it to be. Your arguments do not matter, the number of people that want this do not matter, whatever you think the logical merit of this feature is or why he's wrong for not wanting to add it, it does not matter. If he does not want to add a feature, it's not happening.
> 
> Can someone with the appropriate permissions lock this thread?

Except it stopped being a personal project as soon as Jagex officially endorsed it, get a reality check here bub. There's not even a good reason for it other than what you've stated, "I don't want to and screw anyone that wants it". Pretty piss poor attitude from someone the company behind the game itself is endorsing. Can't wait for the day Jagex finish their own client properly and ban Runelite, it's coming one day, they don't like the lack of control, you know it, I know it, everybody knows it
